meta-client/src/main/java/org/apache/helix/metaclient/impl/zk/ZkMetaClient.java

```diff
@@ -236,6 +239,10 @@ public ConnectState getClientConnectionState() {
   @Override
   public boolean subscribeDataChange(String key, DataChangeListener listener,
       boolean skipWatchingNonExistNode, boolean persistListener) {
+    if (!persistListener) {
+      throw new NotImplementedException("Non-persist (one-time) listener is not supported in ZkMetaClient.");
+    }
+    _zkClient.subscribeDataChanges(key, new DataListenerConverter(listener));
```

Comment:
i may remember it wrong, but i thought there are 2 variations of this method, one explicitly taking listener-type. if so, can we use that variant, please

Reply:
Argh.. we have already additional code.